Github/legos/github_get_branch/github_get_branch.py

```diff
@@ -0,0 +1,74 @@
+##
+##  Copyright (c) 2021 unSkript, Inc
+##  All rights reserved.
+##
+from github import GithubException
+from typing import Optional, List
+from pydantic import BaseModel, Field
+import pprint
+
+
+class InputSchema(BaseModel):
+    branch_name: str = Field(
+        description='Branch name. Eg:"dummy-branch-name"', 
+        title='Branch Name'
+    )
+    owner: str = Field(
+        description='Username of the GitHub user. Eg: "johnwick"',
+        title='Owner'
+    )
+    repository: str = Field(
+        description='Full name of the GitHub repository. Eg: "unskript/Awesome-CloudOps-Automation"',
+        title='Repository',
+    )
+
+
+
+def github_get_branch_printer(output):
+    if not output:
+        return
+    pprint.pprint(output)
+
+def github_get_branch(handle, owner:str, repository: str, branch_name: str) -> List:
+    """github_get_branch returns details of commits (if any) of a branche for a user.
+
+        :type handle: object
+        :param handle: Object returned from task.validate(...).
+
+        :type owner: string
+        :param owner: Username of the GitHub user. Eg: "johnwick"
+
+        :type repository: string
+        :param repository: Full name of the GitHub repository. Eg: "unskript/Awesome-CloudOps-Automation"
+
+        :type branch_name: string
+        :param branch_name: Branch Name Eg: "dummy-branch"
+
+        :rtype: List of branch with commits for a user for a repository
+    """
+    result = []
+    try:
+        user = handle.get_user(login=owner)
```

Comment:
If the user is coming from handle - does that mean that we can only get branches from the repos owned by this account?

What if - for some reason - your development is held up by a PR in another OSS repo?  Could we look for that branch - even if it is owned by another company?

Reply:
I'm seeing 3 API calls here:
Get user, Get repos - get branches.

Since we already KNOW the user, repo and branch, can't we use the API to do just one call to the API?
https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#get-a-branch

Reply:
> What if - for some reason - your development is held up by a PR in another OSS repo? Could we look for that branch - even if it is owned by another company?

We can look up for all orgs/users. But to get teams/members/commits or any other specific info the user with the Personal Access Token should have admin access, otherwise it will throw a 403 error which I have caught.

Reply:
ok.  makes sense